selfdrive/car/tests/test_models.py

```diff
@@ -346,6 +347,11 @@ def test_panda_safety_carstate(self):
       checks['brakePressed'] += brake_pressed != self.safety.get_brake_pressed_prev()
       checks['regenBraking'] += CS.regenBraking != self.safety.get_regen_braking_prev()
 
+      # Verify that panda has the correct velocity for cars that use it (angle based cars)
+      if self.CP.steerControlType in (car.CarParams.SteerControlType.angle,) and not self.CP.notCar:
+        panda_velocity = self.safety.get_vehicle_speed_last() / VEHICLE_SPEED_FACTOR
+        checks['vEgo'] += abs(panda_velocity - CS.vEgoRaw) > 1
+
```

Comment:
we should fix the mismatch instead of increasing the tolerance, this could be the cause of some blocked msgs

Reply:
its a bit awkward because they are in different messages. anything similar you know of? we'd have to store the state of the other message and then update the speed when both are valid